core/src/main/scala/io/finch/request/package.scala

```diff
@@ -34,6 +34,55 @@ import scala.reflect.ClassTag
 package object request {
 
   /**
+   * A reusable validation rule that can be applied to any ''RequestReader'' with a matching type.
+   */
+  class ValidationRule[T] private[request] (val description: String, predicate: T => Boolean) extends (T => Boolean) { self =>
```

Comment:
I actually thought of this myself (I also prefer consistency as you hopefully see in my work). The only reason I went the other way was that I disliked the way you have to avoid having proper parameter names (your choice of `d` above), which makes scaladoc harder to read. If you think this is not a concern, I can change it.